apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -70,18 +106,18 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
     (state: {dance: DanceState}) => state.dance.aiOutput
   );
 
+  // Handle the case in which the modal is created with an existing value.
   useEffect(() => {
     const currentValue = currentAiModalField?.getValue();
-
     if (currentValue) {
-      setMode(Mode.RESULTS_FINAL);
-
-      // The block value will be set to this JSON.
-      setResultJson(currentValue);
-
-      setShowPreview(true);
-
+      setMode(Mode.RESULTS);
       setInputs(JSON.parse(currentValue).inputs);
+      setGeneratingStep([BAD_GENERATED_RESULTS_COUNT, 0]);
```

Comment:
nit: just for readability, could we have some `GOOD_GENERATED_RESULT_INDEX` or `FINAL_GENERATED_RESULT_INDEX` that's just equal to `BAD_GENERATED_RESULTS_COUNT`? I got tripped up with this name until realizing it was being used to index into the last element

Or even, just to be more explicit - what if the type was something like
```
type GeneratedEffects = {
  rejectedEffects: GeneratedEffect[];
  chosenEffect: GeneratedEffect;
}
```
I know there's a lot of assumption with indices here so don't want to disrupt this too much, but just an idea

Reply:
I went ahead and did the extra credit option :)